test/src/test/java/org/corfudb/infrastructure/logreplication/LogReplicationFSMTest.java

```diff
@@ -155,6 +175,128 @@ public void testLogReplicationFSMTransitions() throws Exception {
         transition(LogReplicationEventType.REPLICATION_STOP, LogReplicationStateType.INITIALIZED, true);
     }
 
+    /**
+     * Verify the lastSyncType flag is being initialized properly and SnapshotSyncInfo
+     * reflects accurate sync type for snapshot sync.
+     *
+     */
+    @Test
+    public void testSyncStatusUpdatesForSnapshotOnInit() throws Exception {
+        final int updateToStatusTableFromOnEntry = 1;
+        initLogReplicationFSM(ReaderImplementation.EMPTY);
+
+        final Table<ReplicationStatusKey, ReplicationStatusVal, Message> statusTable =
+                this.corfuStore.getTable(NAMESPACE, REPLICATION_STATUS_TABLE);
+
+        CountDownLatch statusTableLatch = new CountDownLatch(updateToStatusTableFromOnEntry);
+        TestStatusTableStreamListener streamListener = new TestStatusTableStreamListener(statusTableLatch);
+        corfuStore.subscribeListener(streamListener, NAMESPACE, LR_STATUS_STREAM_TAG);
+
+        // Initial state: Initialized
+        LogReplicationState initState = fsm.getState();
+        assertThat(initState.getType()).isEqualTo(LogReplicationStateType.INITIALIZED);
+
+        transitionAvailable.acquire();
+
+        ReplicationStatusVal currentReplicationVal = statusTable.entryStream().findAny().get().getPayload();
+
+        // Default sync value is null so current syncType should not be set.
+        Assert.assertFalse(currentReplicationVal.hasField(ReplicationStatusVal.getDescriptor().findFieldByName("syncType")));
+
+        // Current SyncStatus for ReplicationInfo and SnapshotSyncInfo should be NOT_STARTED
+        Assert.assertEquals(SyncStatus.NOT_STARTED, currentReplicationVal.getStatus());
+        Assert.assertEquals(SyncStatus.NOT_STARTED, currentReplicationVal.getSnapshotSyncInfo().getStatus());
+
+        // Transition #1: SNAPSHOT Sync Start
+        transition(LogReplicationEventType.SNAPSHOT_SYNC_REQUEST, LogReplicationStateType.IN_SNAPSHOT_SYNC);
+
+        // Await for update to status table from entry of InSnapshotSyncState
+        statusTableLatch.await();
+        corfuStore.unsubscribeListener(streamListener);
+
+        Set<SyncType> expectedSyncTypes = new HashSet<>(
+                Collections.singletonList(SyncType.SNAPSHOT));
+        Set<SyncType> actualSyncTypes = new HashSet<>();
+        List<SyncStatus> actualSyncStatus = new ArrayList<>();
+
+        Iterator<CorfuStreamEntries> entriesIterator = streamListener.getEntries().iterator();
```

Comment:
instead of iterating over all values, you can verify that the size of entries is 1 and get that entry.  We expect only 1 update in that transaction so it is a useful validation.